compiler/src/dmd/statement.d

```diff
@@ -1804,6 +1815,12 @@ extern (C++) final class GotoStatement : Statement
             return true;
         }
 
+        if (label.statement.inCtfeBlock && !inCtfeBlock)
+        {
+            error("cannot `goto` into `if (__ctfe)` block");
```

Comment:
Though you can can still goto into an `if (FALSE)` block.

Reply:
`if(FALSE)` is fundamentally different, as it does not change the semantics of the dependent block.